app/src/androidTest/java/Cryptohopper/Android/SDK/exchange/ExchangeInstrumentedNegativeTest.kt

```diff
@@ -0,0 +1,321 @@
+package Cryptohopper.Android.SDK.exchange
+
+import Cryptohopper.Android.SDK.helper.Const
+import Cryptohopper.Android.SDK.helper.Const.API_KEY
+import Cryptohopper.Android.SDK.helper.Const.API_PASSWORD
+import Cryptohopper.Android.SDK.helper.Const.API_USER
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.mervick.aes_everywhere.Aes256
+import cryptohopper.android.sdk.CryptohopperAuth
+import cryptohopper.android.sdk.SharedModels.ConfigModels.HopperAPIEnvironment
+import Cryptohopper.Android.SDK.helper.StringGenerator
+import android.util.Log
+import kotlinx.coroutines.*
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.Before
+
+@DelicateCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class ExchangeInstrumentedNegativeTest {
+
+    @Before
+    fun setup() {
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        CryptoHopperConfig.configure(
+            appContext, API_KEY,
+            HopperAPIEnvironment.Production
+        )
+
+        if (Const.isUserLoggedIn.not()) {
+            callAuthenticationWithMockDetails()
+            Const.isUserLoggedIn = true
+        }
+    }
+
+
+    private fun callAuthenticationWithMockDetails() {
+        val userAgent =
+            Aes256.encrypt(StringGenerator.getRandomString(), StringGenerator.getRandomString())
```

Comment:
This would result in blocking your IP permanently , other tests would fail . AES256 encryption should pass even if your username or password is wrong . So this will effect your other tests due to IP block. Send encryption correctly but send username and password wrong